rankers/SimpleAggregateRanker/__init__.py

```diff
@@ -0,0 +1,35 @@
+from jina.executors.rankers import Chunk2DocRanker
+import numpy as np
+
+class SimpleAggregateRanker(Chunk2DocRanker):
+    """
+    :class:`SimpleAggregateRanker` aggregates the score of the matched doc from the matched chunks.
+    For each matched doc, the score is aggregated from all the matched chunks belonging to that doc.
+    """
+
+    AGGREGATE_FUNCTIONS = ['min', 'max', 'mean', 'median', 'sum', 'prod']
+
+    def __init__(self, aggregate_function: str, is_reversed_score: bool, *args, **kwargs):
+        """
+        :param: aggregate_function: defines the used aggregate function and can be one of:
+        [min, max, mean, median, sum, prod]
+        :param: is_reverse_score: True if a large score of the matched chunks mean a lower score of the matched doc,
+        othewise: False
+
+        """
+        super().__init__(*args, **kwargs)
+        self.is_reversed_score = is_reversed_score
+        if aggregate_function in self.AGGREGATE_FUNCTIONS:
+            self.np_aggregate_function = getattr(np, aggregate_function)
+        else:
+            raise ValueError(f'The aggregate function "{aggregate_function}" is not one of the values: "{self.AGGREGATE_FUNCTIONS}".')
+
+    def _get_score(self, match_idx, query_chunk_meta, match_chunk_meta, *args, **kwargs):
+        scores = match_idx[self.COL_SCORE]
+        aggregated_score = self.np_aggregate_function(scores)
+        if self.is_reversed_score:
+            if aggregated_score < 0:
```

Comment:
I think here we just need to check `DivByZeroError` but raising an exception for something that is well defined should not happen.

Reply:
I made a change accordingly. Now, only zero is forbidden as result of an aggregate function.
However, this will lead to unpredictable behaviour. Let's say we allow `negative distances` in the chunk matches and use the `mean aggregate function.` As a result we would get sometimes positive and sometimes negative doc scores.
So the exception might occur in some rare cases where the mean value actually becomes zero. I would like to avoid this.
Also, the `reversed_score function` is only used for distances and distances can not be smaller than zero.

Reply:
Calling with distances is what would happen most of the time but is not ensured. It could be called with any kind of score. I think the only thing we need to avoid is having a DivByZero which does not happen if aggregate score is 0 right?

Reply:
Yes division by zero does not happen anymore. Now it throws an error.